core/ui.js

```diff
@@ -604,11 +604,74 @@ function onNewMessageNodeAdded(messageNode)
         debugger;
     var msgID = data_id.split("_")[2];
 
+    // check if there are any view-once messages for this message ID
+    // open viewOnce indexedDB
+    restoreViewOnceMessageIfNeeded(msgID, messageNode);
+
     restoreDeletedMessageIfNeeded(messageNode, msgID);
 
     markMessageNodeDeviceIfPossible(messageNode, msgID);
 }
 
+function restoreViewOnceMessageIfNeeded(msgID, messageNode) 
```

Comment:
Let's swap the parameters order here to be similar to `restoreDeletedMessageIfNeeded` and `markMessageNodeDeviceIfPossible`